backend/src/main/java/ch/puzzle/okr/service/KeyResultService.java

```diff
@@ -96,11 +96,16 @@ public List<KeyResultMeasureDto> getAllKeyResultsByObjectiveWithMeasure(Long id)
     @Transactional
     public void deleteKeyResultById(Long id) {
         List<Measure> measures = getAllMeasuresByKeyResult(id);
-        Long objectiveId = getKeyResultById(id).getObjective().getId();
         for (Measure measure : measures) {
             measureRepository.deleteById(measure.getId());
         }
         keyResultRepository.deleteById(id);
+    }
+
+    @Transactional
+    public void deleteKeyResultAndUpdateProgress(Long id) {
+        deleteKeyResultById(id);
+        Long objectiveId = getKeyResultById(id).getObjective().getId();
```

Comment:
Zuerst Löschen und anschliessend lesen ist nicht so die gewünschte Variante, auch wenn dies in diesem Contest wahrscheinlich funktioniert.